pallets/automation-time/src/lib.rs

```diff
@@ -835,19 +870,29 @@ pub mod pallet {
 			action: Action<T>,
 			who: T::AccountId,
 			provided_id: Vec<u8>,
-			time: UnixTime,
+			time: BoundedVec<UnixTime, T::MaxRecurringTimes>,
 		) -> Result<(), Error<T>> {
 			if provided_id.len() == 0 {
 				Err(Error::<T>::EmptyProvidedId)?
 			}
-			Self::is_valid_time(time)?;
+
+			// Sort list of times then remove consecutive duplicates
+			let mut cleaned_times = time.to_vec();
+			cleaned_times.sort_unstable();
+			cleaned_times.dedup();
+			if cleaned_times.len() > T::MaxRecurringTimes::get().try_into().unwrap() {
+				Err(Error::<T>::TooManyRecurringTimes)?;
+			}
+			for scheduled_time in cleaned_times.iter() {
+				Self::is_valid_time(*scheduled_time)?;
+			}
 
 			let fee = Self::calculate_execution_fee(&action);
 			T::NativeTokenExchange::can_pay_fee(&who, fee.clone())
 				.map_err(|_| Error::InsufficientBalance)?;
 
-			let task_id = Self::schedule_task(who.clone(), provided_id.clone(), time)?;
-			let task: Task<T> = Task::<T> { owner_id: who.clone(), provided_id, time, action };
+			let task_id = Self::schedule_task(who.clone(), provided_id.clone(), cleaned_times.clone())?;
```

Comment:
huh i didn't know you could clone vectors. double check that works. but otherwise, might need to do something else here.

Reply:
will definitely double check.  it seems to support it from the docs.  https://doc.rust-lang.org/std/vec/struct.Vec.html#impl-Clone